### PR TITLE
lib.makePerlPath -> perlPackages.makePerlPath

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

First, it has been in TODO comment:
https://github.com/NixOS/nixpkgs/blob/320c9c10de8489604c459127079f59a3ef4d8475/lib/strings.nix#L173-L174

Second, to get rid of hardcoded "lib/perl5/site_perl", because ```perl.libPrefix``` == "lib/perl5/site_perl" only on POSIX platforms, it is not so on Windows

cc @zimbatm

### CODE REVIEW

**Comment by zimbatm** [comment]
`nix-review pr 52100` fails with `OSError: [Errno 7] Argument list too long: 'nix'`. Oops :)

**Comment by volth** [comment]
> Did you test that the touched packages still compile after the change already?

Not every. As ```git``` and ```perl``` are affected it would require full rebuild (what ```staging``` is for :)

File: pkgs/tools/package-management/apt/default.nix [hunk 1]
```diff
@@ -27,15 +26,15 @@ stdenv.mkDerivation rec {
...
   ] ++ lib.optionals withDocs [
-    doxygen Po4a w3m docbook_xml_dtd_45
+    doxygen perlPackages.Po4a w3m docbook_xml_dtd_45
   ] ++ lib.optionals withNLS [
     gettext
   ];
 
   preConfigure = ''
-    export PERL5LIB="$PERL5LIB''${PERL5LIB:+:}${Po4a}/lib/perl5";
+    export PERL5LIB="$PERL5LIB''${PERL5LIB:+:}${perlPackages.Po4a}/lib/perl5";
```
Comment (volth) [inline pkgs/tools/package-management/apt/default.nix]: I am not sure why here is ```lib/perl5``` and not ```lib/perl5/site_perl```
Perhaps it never worked and is not needed at all

Comment (volth) [reply pkgs/tools/package-management/apt/default.nix]: seems like a separated issue, some perl libs (po4a, libapparmor, ...) are installed in ```$out/lib/perl5``` which introduces some irregularity and problems with finding the libs (for example ```${perlPackages.Po4a}/bin/po4a``` does not work looking for the libs in ```${perlPackages.Po4a}/lib/perl5/site_perl``` while they are in ```${perlPackages.Po4a}/lib/perl5```

Comment (volth) [reply pkgs/tools/package-management/apt/default.nix]: https://github.com/NixOS/nixpkgs/pull/52137
I'll cherry-pick it here

**Pull request merged**